alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt

```diff
@@ -507,6 +512,97 @@ class MonitorRunnerIT : AlertingRestTestCase() {
         assertEquals("Enabled times do not match", monitor.enabledTime, retrievedMonitor.enabledTime)
     }
 
+    fun `test monitor with throttled action for same alert`() {
+        val actionThrottleEnabled = randomAction(template = randomTemplateScript("Hello {{ctx.monitor.name}}"),
+                destinationId = createDestination().id,
+                throttleEnabled = true, throttle = Throttle(value = 5, unit = MINUTES))
+        val actionThrottleNotEnabled = randomAction(template = randomTemplateScript("Hello {{ctx.monitor.name}}"),
+                destinationId = createDestination().id,
+                throttleEnabled = false, throttle = Throttle(value = 5, unit = MINUTES))
+        val actions = listOf(actionThrottleEnabled, actionThrottleNotEnabled)
+        val monitor = createMonitor(randomMonitor(triggers = listOf(randomTrigger(condition = ALWAYS_RUN, actions = actions)),
+                schedule = IntervalSchedule(interval = 1, unit = ChronoUnit.MINUTES)))
+
+        verifyActionThrottleResults(monitor.id, mutableMapOf(Pair(actionThrottleEnabled.id, false),
+                Pair(actionThrottleNotEnabled.id, false)))
+
+        val alerts1 = searchAlerts(monitor)
+        assertEquals("1 alert should be returned", 1, alerts1.size)
+        verifyAlert(alerts1.single(), monitor, ACTIVE)
+        val actionResults1 = verifyActionExecutionResultInAlert(alerts1[0],
+                mutableMapOf(Pair(actionThrottleEnabled.id, 0), Pair(actionThrottleNotEnabled.id, 0)))
+
+        assertEquals(actionResults1.size, 2)
+        verifyActionThrottleResults(monitor.id, mutableMapOf(Pair(actionThrottleEnabled.id, true),
```

Comment:
Let's move executeMonitor on here so that it is easy to understand.

Reply:
Thanks, change it